tests/testthat/test-format_data.R

```diff
@@ -1,442 +1,162 @@
-    DRUG = rep("DrugA", 20),
-    PARAM = rep("Analyte1", 20),
-    AFRLT = rep(seq(0, 9), 2),
-    ARRLT = rep(seq(0, 4), 4),
-    NFRLT = rep(seq(0, 9), 2),
-    DOSNO = rep(1, 20),
-    ROUTE = "extravascular",
-    DOSEA = 10,
-    ADOSEDUR = 0,
-    AVAL = runif(20)
-  )
-
-  # Call format_pkncaconc_data
-  df_conc <- format_pkncaconc_data(ADNCA,
-                                   group_columns = c("STUDYID", "USUBJID", "PCSPEC",
-                                                     "DRUG", "PARAM"),
-                                   time_column = "AFRLT")
-
-  # Call format_pkncadose_data
-  df_dose <- format_pkncadose_data(df_conc,
-                                   group_columns = c("STUDYID", "USUBJID", "PCSPEC",
-                                                     "DRUG", "PARAM"),
-                                   time_column = "AFRLT")
-
-  # Generate PKNCAconc and PKNCAdose objects
-  pknca_conc <- PKNCA::PKNCAconc(
-    df_conc,
-    formula = AVAL ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID / PARAM,
-    exclude_half.life = "exclude_half.life",
-    time.nominal = "NFRLT"
-  )
-
-  pknca_dose <- PKNCA::PKNCAdose(
-    data = df_dose,
-    formula = DOSEA ~ TIME_DOSE | STUDYID + PCSPEC + DRUG + USUBJID,
-    route = "ROUTE",
-    time.nominal = "NFRLT",
-    duration = "ADOSEDUR"
-  )
-
-  # Define the other arguments for the dose intervals
-  params <- c("cmax", "tmax", "half.life", "cl.obs")
-
-  # Call format_pkncadata_intervals
-  myintervals <- format_pkncadata_intervals(pknca_conc,
-                                            pknca_dose,
-                                            params = params,
-                                            start_from_last_dose = TRUE)
-
-  # Test if myintervals is a data frame
-  expect_s3_class(myintervals, "data.frame")
-
-  # Test if myintervals has the correct columns
-  expect_true(all(c("start", "end", "STUDYID",
-                    "USUBJID", "DOSNOA", "type_interval") %in% colnames(myintervals)))
-
-  # Test if interval types are classified as "main"
-  expect_true(all(myintervals$type_interval == "main"))
-
-  # Test if myintervals can be used with PKNCAdata by testing its output
-  expect_no_error(
-    PKNCA::PKNCAdata(
-      data.conc = pknca_conc,
-      data.dose = pknca_dose,
-      intervals = myintervals,
-      units = PKNCA::pknca_units_table(
-        concu = pknca_conc$data$PCSTRESU[1],
-        doseu = pknca_conc$data$DOSEU[1],
-        amountu = pknca_conc$data$PCSTRESU[1],
-        timeu = pknca_conc$data$RRLTU[1]
+                                                       "DRUG", "PARAM"),
+                                     time_column = "AFRLT")
+
+    expect_s3_class(df_dose, "data.frame")
+    expect_true(all(c("STUDYID", "USUBJID", "PCSPEC",
+                      "DRUG", "PARAM", "AFRLT",
+                      "ARRLT", "AVAL", "conc_groups",
+                      "TIME", "TIME_DOSE", "DOSNOA", "ROUTE") %in% colnames(df_dose)))
```

Comment:
nitpick:
```suggestion
    expect_setequal(c("STUDYID", "USUBJID", "PCSPEC",
                      "DRUG", "PARAM", "AFRLT",
                      "ARRLT", "AVAL", "conc_groups",
                      "TIME", "TIME_DOSE", "DOSNOA", "ROUTE"), colnames(df_dose))
```
as before, this is assuming the are expecting the exact set of columns and nothing more.